### ISSUE TITLE
Installation failed

### ISSUE THREAD

**Issue by monkey-Al** [Alfred-hhy/fangmeituan#1]
G:\courses\3-1\openSource\homework\pr\test\vue_project>npm i
npm WARN old lockfile
npm WARN old lockfile The package-lock.json file was created with an old version of npm,
npm WARN old lockfile so supplemental metadata must be fetched from the registry.
npm WARN old lockfile
npm WARN old lockfile This is a one-time fix-up, please be patient...
npm WARN old lockfile
npm WARN old lockfile vue-loader-v16: No matching version found for vue-loader-v16@16.8.3.
npm WARN old lockfile     at module.exports (G:\software\nvm\nvm\v18.17.1\node_modules\npm\node_modules\npm-pi
ck-manifest\lib\index.js:209:23)
npm WARN old lockfile     at RegistryFetcher.manifest (G:\software\nvm\nvm\v18.17.1\node_modules\npm\node_modu
les\pacote\lib\registry.js:119:22)
npm WARN old lockfile     at async Array.<anonymous> (G:\software\nvm\nvm\v18.17.1\node_modules\npm\node_modul
es\@npmcli\arborist\lib\arborist\build-ideal-tree.js:727:24)
npm WARN old lockfile  Could not fetch metadata for vue-loader-v16@16.8.3 vue-loader-v16: No matching version 
found for vue-loader-v16@16.8.3.
npm WARN old lockfile     at module.exports (G:\software\nvm\nvm\v18.17.1\node_modules\npm\node_modules\npm-pi
ck-manifest\lib\index.js:209:23)
npm WARN old lockfile     at RegistryFetcher.manifest (G:\software\nvm\nvm\v18.17.1\node_modules\npm\node_modu
les\pacote\lib\registry.js:119:22)
npm WARN old lockfile     at async Array.<anonymous> (G:\software\nvm\nvm\v18.17.1\node_modules\npm\node_modul
es\@npmcli\arborist\lib\arborist\build-ideal-tree.js:727:24) {
npm WARN old lockfile   code: 'ETARGET',
npm WARN old lockfile   type: 'version',
npm WARN old lockfile   wanted: '16.8.3',
npm WARN old lockfile   versions: [
npm WARN old lockfile     '16.0.0-beta.5.4',
npm WARN old lockfile     '16.0.0-beta.5.3',
npm WARN old lockfile     '16.0.0-beta.5.2',
npm WARN old lockfile     '16.0.0-beta.5.1',
npm WARN old lockfile     '16.0.0-beta.5'
npm WARN old lockfile   ],
npm WARN old lockfile   distTags: { latest: '16.0.0-beta.5.4' },
npm WARN old lockfile   defaultTag: 'latest'
npm WARN old lockfile }
npm WARN deprecated @babel/plugin-proposal-private-methods@7.16.7: This proposal has been merged to the ECMASc
ript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-private-methods
 instead.
npm WARN deprecated @babel/plugin-proposal-unicode-property-regex@7.16.7: This proposal has been merged to the
 ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-unicode-
property-regex instead.
npm WARN deprecated @babel/plugin-proposal-private-property-in-object@7.16.7: This proposal has been merged to
 the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-priv
ate-property-in-object instead.
npm WARN deprecated @babel/plugin-proposal-optional-catch-binding@7.16.7: This proposal has been merged to the
 ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional
-catch-binding instead.
npm WARN deprecated @babel/plugin-proposal-numeric-separator@7.16.7: This proposal has been merged to the ECMA
Script standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-numeric-separ
ator instead.
npm WARN deprecated @babel/plugin-proposal-nullish-coalescing-operator@7.16.7: This proposal has been merged t
o the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-nul
lish-coalescing-operator instead.
npm WARN deprecated @babel/plugin-proposal-logical-assignment-operators@7.16.7: This proposal has been merged 
to the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-lo
gical-assignment-operators instead.
npm WARN deprecated @babel/plugin-proposal-dynamic-import@7.16.7: This proposal has been merged to the ECMAScr
ipt standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-dynamic-import i
nstead.
npm WARN deprecated @babel/plugin-proposal-export-namespace-from@7.16.7: This proposal has been merged to the 
ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-export-na
mespace-from instead.
npm WARN deprecated @babel/plugin-proposal-class-properties@7.16.7: This proposal has been merged to the ECMAS
cript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-properti
es instead.
npm WARN deprecated @babel/plugin-proposal-class-static-block@7.16.7: This proposal has been merged to the ECM
AScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-class-static
-block instead.
npm WARN deprecated @babel/plugin-proposal-json-strings@7.16.7: This proposal has been merged to the ECMAScrip
t standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-json-strings inste
ad.
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library 
is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Ref
erence/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated source-map-url@0.4.1: See https://github.com/lydell/source-map-url#deprecated
npm WARN deprecated eslint-loader@2.2.1: This loader has been deprecated. Please use eslint-webpack-plugin    
npm WARN deprecated @hapi/bourne@1.3.2: This version has been deprecated and is no longer supported or maintai
ned
npm WARN deprecated @hapi/topo@3.1.6: This version has been deprecated and is no longer supported or maintaine
d
npm WARN deprecated @babel/plugin-proposal-async-generator-functions@7.16.8: This proposal has been merged to 
the ECMAScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-async
-generator-functions instead.
npm WARN deprecated @babel/plugin-proposal-optional-chaining@7.16.7: This proposal has been merged to the ECMA
Script standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-optional-chai
ning instead.
npm WARN deprecated consolidate@0.15.1: Please upgrade to consolidate v1.0.0+ as it has been modernized with s
everal long-awaited fixes implemented. Maintenance is supported by Forward Email at https://forwardemail.net ;
 follow/watch https://github.com/ladjs/consolidate for updates and release changelog
npm WARN deprecated urix@0.1.0: Please see https://github.com/lydell/urix#deprecated
npm WARN deprecated har-validator@5.1.5: this library is no longer supported
npm WARN deprecated @babel/plugin-proposal-object-rest-spread@7.16.7: This proposal has been merged to the ECM
AScript standard and thus this plugin is no longer maintained. Please use @babel/plugin-transform-object-rest-
spread instead.
npm WARN deprecated source-map-resolve@0.5.3: See https://github.com/lydell/source-map-resolve#deprecated     
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokid
ar 3 with 15x fewer dependencies
npm WARN deprecated chokidar@2.1.8: Chokidar 2 does not receive security updates since 2019. Upgrade to chokid
ar 3 with 15x fewer dependencies
npm WARN deprecated resolve-url@0.2.1: https://github.com/lydell/resolve-url#deprecated
npm WARN deprecated sourcemap-codec@1.4.8: Please use @jridgewell/sourcemap-codec instead
npm WARN deprecated html-webpack-plugin@3.2.0: 3.x is no longer supported
npm WARN deprecated @hapi/address@2.1.4: Moved to 'npm install @sideway/address'
npm WARN deprecated querystring@0.2.0: The querystring API is considered Legacy. new code should use the URLSe
archParams API instead.
npm WARN deprecated babel-eslint@10.1.0: babel-eslint is now @babel/eslint-parser. This package will no longer
 receive updates.
npm WARN deprecated uuid@3.4.0: Please upgrade  to version 7 or higher.  Older versions may use Math.random() 
in certain circumstances, which is known to be problematic.  See https://v8.dev/blog/math-random for details. 
npm WARN deprecated request@2.88.2: request has been deprecated, see https://github.com/request/request/issues
/3142
npm WARN deprecated @hapi/hoek@8.5.1: This version has been deprecated and is no longer supported or maintaine
d
npm WARN deprecated @hapi/joi@15.1.1: Switch to 'npm install joi'
npm WARN deprecated svgo@1.3.2: This SVGO version is no longer supported. Upgrade to v2.x.x.
npm WARN deprecated core-js@3.20.3: core-js@<3.23.3 is no longer maintained and not recommended for usage due 
to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause
 a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upg
rade your dependencies to the actual version of core-js.

added 1379 packages in 32s

G:\courses\3-1\openSource\homework\pr\test\vue_project>npm run serve

> vue_project@0.1.0 serve
> vue-cli-service serve

Browserslist: caniuse-lite is outdated. Please run:
  npx browserslist@latest --update-db
  Why you should do it regularly: https://github.com/browserslist/browserslist#browsers-data-updating
 INFO  Starting development server...
10% building 2/4 modules 2 active ...server\client\index.js?http://192.168.236.136:8080&sockPath=/sockjs-nodeE
rror: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:69:19)
    at Object.createHash (node:crypto:133:10)
    at module.exports (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\webpack\lib\util\cr
eateHash.js:135:53)
    at NormalModule._initBuildHash (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\webpac
k\lib\NormalModule.js:417:16)
    at handleParseError (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\webpack\lib\Norma
lModule.js:471:10)
    at G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\webpack\lib\NormalModule.js:503:5  
    at G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\webpack\lib\NormalModule.js:358:12 
    at G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\loader-runner\lib\LoaderRunner.js:3
73:3
    at iterateNormalLoaders (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\loader-runner
\lib\LoaderRunner.js:214:10)
    at iterateNormalLoaders (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\loader-runner
\lib\LoaderRunner.js:221:10)
    at G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\loader-runner\lib\LoaderRunner.js:2
36:3
    at runSyncOrAsync (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\loader-runner\lib\L
oaderRunner.js:130:11)
    at iterateNormalLoaders (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\loader-runner
\lib\LoaderRunner.js:232:2)
    at Array.<anonymous> (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\loader-runner\li
b\LoaderRunner.js:205:4)
    at Storage.finished (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\enhanced-resolve\
lib\CachedInputFileSystem.js:55:16)
    at G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\enhanced-resolve\lib\CachedInputFil
eSystem.js:91:9
node:internal/crypto/hash:69
  this[kHandle] = new _Hash(algorithm, xofLen);
                  ^

Error: error:0308010C:digital envelope routines::unsupported
    at new Hash (node:internal/crypto/hash:69:19)
    at Object.createHash (node:crypto:133:10)
    at module.exports (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\webpack\lib\util\cr
eateHash.js:135:53)
    at NormalModule._initBuildHash (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\webpac
k\lib\NormalModule.js:417:16)
    at handleParseError (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\webpack\lib\Norma
lModule.js:471:10)
    at G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\webpack\lib\NormalModule.js:503:5  
    at G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\webpack\lib\NormalModule.js:358:12 
    at G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\loader-runner\lib\LoaderRunner.js:3
73:3
    at iterateNormalLoaders (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\loader-runner
\lib\LoaderRunner.js:214:10)
    at Array.<anonymous> (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\loader-runner\li
b\LoaderRunner.js:205:4)
    at Storage.finished (G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\enhanced-resolve\
lib\CachedInputFileSystem.js:55:16)
    at G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\enhanced-resolve\lib\CachedInputFil
eSystem.js:91:9
    at G:\courses\3-1\openSource\homework\pr\test\vue_project\node_modules\graceful-fs\graceful-fs.js:123:16  
    at FSReqCallback.readFileAfterClose [as oncomplete] (node:internal/fs/read_file_context:68:3) {
  opensslErrorStack: [ 'error:03000086:digital envelope routines::initialization error' ],
  library: 'digital envelope routines',
  reason: 'unsupported',
  code: 'ERR_OSSL_EVP_UNSUPPORTED'
}

Node.js v18.17.1

G:\courses\3-1\openSource\homework\pr\test\vue_project>

The above is the command-line installation process I followed based on your instructions, but I encountered an error when running the 'npm run serve' command. How should I resolve this issue?

**Comment by Alfred-hhy** [comment]
What version of Node.js are you using? It might be the reason for the issue.

**Comment by monkey-Al** [comment]
I'm using version 18.17.1. What version are you using? Do I have to match your version exactly? If so, wouldn't that cause compatibility issues for different projects? It seems a bit cumbersome to change versions every time.

**Comment by Alfred-hhy** [comment]
Hmm, I'm using version 14.17.6. I'll run some tests on your question shortly and provide you with a better answer tonight.

**Comment by monkey-Al** [comment]
I'm really grateful for your help. Thank you so much.

**Comment by Alfred-hhy** [comment]
I've verified your issue, and indeed, the project doesn't support Node.js 18. It should work fine with version 14. Regarding the difficulty in managing Node.js versions, I suggested using the nvm tool. I use it myself and find it very convenient.

https://zhuanlan.zhihu.com/p/337663452    in Chinese
https://www.freecodecamp.org/news/nvm-for-windows-how-to-download-and-install-node-version-manager-in-windows-10/  
 in English
